Backend/Tumultu/Domain/Entities/FileEntity.cs

```diff
@@ -5,11 +5,42 @@ namespace Tumultu.Domain.Entities;
 
 public class FileEntity : BaseEntity<Guid>
 {
-    public IEnumerable<FileVariant> Variants { get; set; } = [];
-    public IEnumerable<FileString> FileStrings { get; set; } = [];
+    private FileEntity(string md5, string sha1, string sha256, byte[] payload)
+    {
+        this.MD5Signature = md5;
+        this.SHA1Signature = sha1;
+        this.SHA256Signature = sha256;
+        this.Size = payload.Length;
+    }
+    public IList<FileVariant> Variants { get; set; } = [];
+    public IList<FileString> FileStrings { get; set; } = [];
     public int Size { get; set; } = -1;
     public string? MD5Signature { get; set; }
     public string? SHA1Signature { get; set; }
     public string? SHA256Signature { get; set; }
     public AnalysisResult? AnalysisResult { get; set; }
+
+    public static async Task<FileEntity> CreateAsync(string md5, string sha1, string sha256, byte[] payload, User uploadingUser, CancellationToken cancellationToken)
```

Comment:
I wonder if async method variants are really needed in the domain layer.

Reply:
To be discussed I guess, I don't really see a problem with it, but I am curious why shouldn't we